### PR TITLE
Add status change confirmation

### LINKED ISSUE #792: Status changes should have a confirmation dialog
Has been requested that a confirmation dialog is shown before statuses are changed.

I think this is likely a common thing. So.. generic?!

Thinking a hook concept similar to `useNotification` where we can pass a message and callback for the `ok`.

An api I think would be good would be something like..

```
const { onGetConfirmation } = useConfirmation({
    message: "This will change the status to Verified. Are you sure?",
    onConfirm: () => // do the update
})
```

1. the `onGetConfirmation` naming is poor
2. The message could have some nicer dx. passing a label, or not having to put "Are you sure?" for example.
3. I think an "Are you sure?" confirmation is pretty standard. Are there other variants of this type of modal? The `useNofications` is awesome because you can choose a variant but have the same api, so if there was different variants of this problem, we could make this hook have multiple variants also. Maybe too far.

An alternative api could be something promise based. I think this might be a little harder to implement. Though I think the api is quite nice to use 🤔 

```
const { open } = useConfirmationModal({
    message: "This will do things...."
})

...

onClick={async () => {
    const result = await open();
    if (result.ok) // update
}}

```

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #792 

- This implements the confirmation for inbound shipments confirmation modal

### CODE REVIEW

**Review by joshxg — COMMENTED** [review]
I've not had a chance to test this on the real api yet as there are a lot of changes to make after merges which i'll look at next

File: packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  }, [status, options]);
+
+  return { options, selectedOption, setSelectedOption, onGetConfirmation };
+};
+
+export const StatusChangeButton = () => {
+  const { options, selectedOption, setSelectedOption, onGetConfirmation } =
+    useStatusChangeButton();
+  const isDisabled = !useIsInboundEditable();
+  console.log(isDisabled, selectedOption);
```
Comment (mark-prins) [inline packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx:134]: debug code?

File: packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  if (currentStatus === InvoiceNodeStatus.New) {
+    // When the status is new, delivered and verified are available to
+    // select.
+    options[3].isDisabled = false;
+    options[4].isDisabled = false;
+  }
+
+  // When the status is delivered, only verified is available to select.
+  if (currentStatus === InvoiceNodeStatus.Delivered) {
+    options[3].isDisabled = false;
```
Comment (mark-prins) [inline packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx:61]:
```suggestion
    options[4].isDisabled = false;
```

**Review by mark-prins — APPROVED** [review]
🎉  awesome!

File: packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx (hunk 1 above)
Comment (joshxg) [reply packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx:134]: 😁

File: packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx (hunk 2 above)
Comment (joshxg) [reply packages/invoices/src/InboundShipment/DetailView/Footer/StatusChangeButton.tsx:61]: argh! thanks

**Pull request merged**